### PR TITLE
fix: TransactionStatus::status should not be waiting for Final

### PR DESCRIPTION
**Pull request by ChaoticTempest**
The `TransactionStatus::status` call should not be waiting for `Final` inclusion of the transaction as it is supposed to only be checking the status of the transaction on chain and not wait for it to completely complete. Also, this does not take away any feature as `TransactionStatus::wait` already does the full wait

### CODE REVIEW

File: workspaces/src/operations.rs
Comment (frol) [inline workspaces/src/operations.rs:522]: Indeed, final is not needed, but looking at the implementation, it feels that you actually need ExecutedOptimistic since the function needs to return the transaction outcome, right?

Comment (ChaoticTempest) [reply workspaces/src/operations.rs:522]: nah, this is just polling the transaction status, so ExecutedOptimistic would have the transaction already be slightly  completed. This function should just be returning immediately where ExecutedOptimistic would take some time to resolve

Comment (frol) [reply workspaces/src/operations.rs:522]: Ah, I misinterpreted `Poll<>` part in the function signature. I thought about it as part of the native Rust futures polling, but it is not.

**Review by frol — APPROVED** [review]
@race-of-sloths score 3

**Comment by race-of-sloths** [comment]
@frol Thank you for calling!
    
@ChaoticTempest Thank you for the contribution! Join [Race of Sloths](https://race-of-sloths.com) by simply mentioning me in your comment/PRs description and start collecting Sloth Points through contributions to open source projects.

<details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/ChaoticTempest)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Pull request merged**
